### PR TITLE
Add CustomStatus class and document it

### PR DESCRIPTION
**Pull request by apple502j**
### Summary
Add CustomStatus class that represents custom status feature. This is a slimmed down version of Activity.

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Rapptz** [comment]
It's too early for this feature since it's subject to change.

Also, there is the fact that there's the guild/channel specific thing that is bound to change.

**Comment by khazhyk** [comment]
could we merge the enum so that stuff like Activity.to_dict() isn't broken

```
Traceback (most recent call last):
  File "/var/lib/spoobeta/dango.py/venv/lib/python3.5/site-packages/discord/activity.py", line 166, in to_dict
    ret['type'] = self.type.value
AttributeError: 'int' object has no attribute 'value'
```

**Comment by Rapptz** [comment]
This was a silent change in the 1.2.3 release: https://github.com/Rapptz/discord.py/commit/1e982e0042ae6c50e59b9b77fb77aa6f361ef0e9

Update your library.

**Comment by OrangutanGaming** [comment]
Custom status is now released
Also, will this be added to self bots or are they feature locked?

File: discord/activity.py [hunk 1]
```diff
@@ -582,6 +583,77 @@ def party_id(self):
...
+
+    @property
+    def status(self):
+        """:class:`str` : The custom status.
+
+        There is an alias for this named :meth:`state`."""
+        return self.state
+
+    def __eq__(self, other):
+        return (isinstance(other, CustomStatus) and other._state === self._state)
```
Comment (algmyr) [inline discord/activity.py]: === is not valid python

**Comment by apple502j** [comment]
Closing this because Wumpus asked me to do

**Pull request closed without merging**